src/hotspot/share/memory/arena.cpp

```diff
@@ -91,9 +91,23 @@ class ChunkPool {
     _first = nullptr;
   }
 
+  bool empty() const {
+    return _first == nullptr;
+  }
+
+  static bool needs_cleaning() {
+    STATIC_ASSERT(_num_pools == 4);
+    return !_pools[0].empty() || !_pools[1].empty() ||
+           !_pools[2].empty() || !_pools[3].empty();
+  }
+
   static void clean() {
-    for (int i = 0; i < _num_pools; i++) {
-      _pools[i].prune();
+    ThreadCritical tc;
```

Comment:
Why do you need `ThreadCritical` here? I would have thought `PauseMark` handles everything right?